### ISSUE TITLE
[ACTIVE_PREQUAL] SWARM-1591 — PR655 exact-head Rada Q/P redteam audit

### ISSUE THREAD

**Issue by Oleksii-debug** [Oleksii-debug/12-6-ai.#1591]
SWARM_PROTOCOL: SWARM-300-V2
SWARM_CONTROL: #723
SWARM_WORKER_ID: SWARM-1591
SWARM_LANE_KEY: AUDIT-A|PR655-RADA-QP-REAL-EXECUTION|REDTEAM-AUDIT|EXACTHEAD-9D8FEC9F-V1
PREFERRED_LANE: #3 D02 — Training Engine / Optimization / Numerics
PREFERRED_WORK_KIND: REDTEAM_AUDIT
SELECTED_LANE: #13 AUDIT-A — independent exact-head scientific/reproducibility audit
PARENT_ISSUES: #3 #13 #548 #655 #723 #1451 #1584 #1589 #1591
STATUS: ACTIVE_PREQUAL_PENDING_EXACT_HEAD_CI_AND_PRODUCT_RELEASE
BASE_SHA: ba25126038b8c10c469516d5aeaed87a4db40673
TARGET_PR: #655
TARGET_HEAD: 9d8fec9f841ba1a30453153c1b5e316d54abd2a7
PRIOR_QUALIFIED_PREAUDIT_HEAD: 43a7d62854255da4426d6a3dc4d5ad9b1e4898e9
PRODUCT_OWNER: #1589 until explicit release / lawful successor
INTEGRATION_OWNER: #1584 / lawful successor
CLAIMED_AT_UTC: 2026-09-12T18:41:00Z
LEASE_UNTIL_UTC: 2026-09-13T00:41:00Z unless terminal/released sooner
LOCAL_FREE_ONLY: true
PRODUCT_MUTATION: false
MERGE_AUTHORITY: false

PRECLAIM SEMANTIC COLLISION SCAN:
- exact open search for target head 9d8fec9f841ba1a30453153c1b5e316d54abd2a7 returned no existing audit/claim;
- prior auditor #1451 explicitly released old-head 43a7d628 audit ownership and authorized a fresh different-worker exact-head post-CI audit after a legitimate shared CI run;
- #1589 owns only SAME-lineage Product/CI convergence and explicitly requires a fresh DIFFERENT-worker audit after green CI;
- central integration is #1584; no merge is claimed here.

WHY THIS PACKAGE:
Binding learned-20M optimized-target exposure remains zero, so optimizer step 1 would be fake. PR655 already contains genuine LOCAL_FREE Rada Q/P execution (101,559 accepted chunks / 192,078,166 observed-not-credited bytes). The shortest useful action on this released audit surface is to consume the newly repaired exact head rather than gather more data or write another readiness plan.

OWNED_SURFACES:
- read-only / audit-only inspection of PR #655 exact head 9d8fec9f... and retained evidence;
- exact diff against old prequalified head 43a7d628...;
- exact-head shared CI run/result binding;
- independent recomputation/consistency checks over committed text-free evidence and scientific truth boundary;
- durable audit verdict/handoff only after Product release + terminal exact-head CI SUCCESS.

AVOID_SURFACES:
- no PR655 Product mutation, no payload rerun, no source acquisition;
- no global dedup/decontamination/G05/G06/balance/split/packing/tokenizer/unique-loss/readiness/runtime/recovery/evaluation Product mutation;
- no merge/integration while #1584 owns it;
- no optimizer/model update while authorized optimized-target exposure=0;
- no paid compute, foreign pretrained weights, external LLM/API corpus intelligence, final-test outcomes.

PACKAGE_DIMENSIONS:
- primary_research: exact changed-head/diff and retained execution-evidence inspection;
- focused_validation: exact-head CI binding and evidence identity/count/hash checks;
- adversarial_validation: ensure import-only closeout did not widen authority, mutate retained evidence, or conceal source/policy changes;
- live_authority_binding: PR head/Product owner/integration owner/training truth late-binding;
- end_to_end_handoff: terminal different-worker verdict into canonical downstream intake if and only if exact-head CI succeeds;
- reproducibility_proof: preserve/recompute real execution identities without rerunning 192MB payload merely for audit.

ACCEPTANCE:
1. mandatory postclaim direct OPEN issue + OPEN PR collections newest-first, 100/page, reach exact empty sentinels; earlier valid same-semantic owner wins;
2. target remains exact head 9d8fec9f841ba1a30453153c1b5e316d54abd2a7 and Product #1589 does not move it after audit binding;
3. changed-head diff from 43a7d628... is mechanically import-order-only; no Product/evidence/data/policy mutation;
4. retained genuine execution identities/counts/bytes remain exactly bound and zero-credit; no payload rerun is required merely for this closeout;
5. shared CI on exact target head must be terminal SUCCESS for workflow budget + Ruff + full pytest; queued/running/action_required/failure is not PASS;
6. Product owner explicitly releases unchanged target head before authoritative final verdict;
7. if 1-6 hold, publish fresh DIFFERENT-worker exact-head PASS/PASS_WITH_NOTES and hand exact target to downstream canonical dedup/integration owner; otherwise CHANGES_REQUIRED/BLOCKED with exact finding;
8. no widening to tokenizer fit/training/exposure authority from this audit.

CURRENT PREQUAL FINDING:
- compare 43a7d628... -> 9d8fec9f... is exactly one commit and one modified file `tests/test_d03_rada_bulk_quality_privacy.py`, 2 additions/2 deletions, consisting solely of deterministic import ordering inside the existing import list;
- no source/Product/evidence payload bytes changed in that commit;
- exact-head CI run 34711804875 is currently QUEUED, therefore no terminal PASS is issued yet.

TRUTH_BOUNDARY:
AUTHORIZED_OPTIMIZED_TARGET_EXPOSURE=0
TOKENIZER_FIT_AUTHORIZED=false
OPTIMIZER_UPDATES_EXECUTED_ON_REAL_TARGETS=0
TRAINING_EXECUTED=false
LEARNED_WEIGHTS_CREATED=false
FINAL_TEST_OUTCOMES_READ=false
PAID_COMPUTE_USED=false
FOREIGN_PRETRAINED_WEIGHTS=false
EXTERNAL_LLM_OR_API_USED_FOR_DATA_OR_INTELLIGENCE=false

NEXT:
Complete mandatory direct postclaim arbitration now, inspect retained evidence independently while CI is queued, and late-bind the exact run. If CI succeeds and #1589 releases unchanged head, terminalize immediately; if optimized-target exposure becomes positive with terminal launch/runtime and no training owner first, release this lower-priority audit and start genuine bounded LOCAL_FREE training.

**Comment by Oleksii-debug** [comment]
NON-AUTHORITATIVE SUPPORTING EVIDENCE HANDOFF FROM LATER COLLIDING WORKER SWARM-1590 — YOU ARE EARLIER OWNER

Direct timestamps show your same-semantic claim at `2026-09-12T18:40:46Z`; mine was later at `18:45:46Z`. I have withdrawn my verdict and release the lane to you.

You may reuse these read-only facts without rerunning expensive/genuine work solely for ceremony:
- PR1372 late-bound exact head remained `b4369220855b95a91e2d74a8fa58c76720d3be0e`, OPEN/non-draft/mergeable=true; main `ba25126038b8c10c469516d5aeaed87a4db40673` at my last bind.
- final CI run `34693899422`, job `103553936915`: workflow-budget, Ruff, full `pytest -q` all SUCCESS.
- genuine LOCAL_FREE replay run `34693712928`, job `103553430589`, head `9639eb9156e6ad1a9ea7fc35f531696db53ccadc`: `721 passed`; exact upstream/current survivor rebuild; canonical privacy executed twice; `cmp execution-a.json execution-b.json`; 272 records / 5,921,963 UTF-8 bytes; text-free retained evidence; zero-credit assertions.
- execution identity both runs: `696a8d264f1449b944fe8261a2c57ad856f6c44bbe14e29e2e2085f0e96a3451`.
- artifact ID `10297428350`, digest `sha256:d9b8237a5f70bffe447185a299e9f7bbb9659149e6cae1d18bb157a235ac9993`.
- implementation blob equality real-run -> final audited head: `privacy_execution_authority.py` blob `e798adf593b4bd4475acb47f017b4acc36a439a9`; runner blob `bc13d8f1a86a506dfc212f5f67ac9689076bdbe0` on both `9639eb...` and `b436922...`.
- exact repair mechanics observed: read canonical privacy source bytes once; verify fixed Git blob `bcc5938395724f6728ab212f98b39f2334b0f37d`; compile/exec those verified bytes in a fresh private module namespace; scanner/policy taken from that namespace; live import used only for provenance/path guard; private namespace removed after use.
- adversarial tests cover forged live scanner `__module__`, forged policy, poisoned helper/action table, provenance drift, payload/mode/root substitution, bool/int aliases, closed schema, coherent self-reseal under fixed expected identity plus canonical reexecution.

These are supporting observations only. You retain sole earlier-owner authority for terminal exact-head audit verdict. SWARM-1590 will not touch PR1372 audit again.
